Исправление ошибок

NLazutkin opened the pull request:
Исправление ошибок в приложении в рамках подготовки к слиянию веток main и develop

NLazutkin committed:
rev. 1.5 Реализация функционала "Отзывы" и "Оценки отзывов".
- В контроллерах и классах query добавлена аннотация @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
- Во всем проекте записи в лог с помощью String.Format исправлены на параметризованный тип записи из методов логгера.

NLazutkin committed:
rev. 1.6 Реализация функционала "Отзывы" и "Оценки отзывов".
- Откат изменений связанных с  аннотацией @FieldDefaults для контроллеров, т.к. это вызывает ошибки при в работе аннотаций типа @...Mapping

NLazutkin committed:
Merge branch 'develop' of https://github.com/NLazutkin/java-filmorate into add-reviews

NLazutkin committed:
Merge branch 'develop' into add-reviews

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/service/UserService.java

NLazutkin committed:
rev. 1.7 Исправление ошибок в приложении в рамках подготовки к слиянию веток main и develop
Так же доработано:
- enums для "Ленты событий" вынесены в отдельный пакет
- FeedService в метод getUserFeed() добавлена проверка пользователя
- Сортировка по рейтингу полезности для списка отзывов

Рефакторинг:
- Запросов для нахождения популярных фильмов
- Метода update класса FilmService
- Формат сообщений в логах
- Добавлены аннотации @Positive и @NotBlank в модели
- Снят ПК с таблицы likes, добавлено свойство для таблицы films_directors ON DELETE CASCADE

Промежуточные таблицы записей о режиссерах\жанрах:
- Рефакторинг добавления
- Добавление методов удаления
- Добавлены SQL запросы

NLazutkin committed:
rev. 1.8 Исправление ошибок в приложении в рамках подготовки к слиянию веток main и develop

Добавлены:
- Тесты методов для модели Director
- Тесты хранилища и валидации для модели Feed

Рефакторинг:
- Разделены data.sql для приложения и тестов с разным набором данных

dante-must-die reviewed (dismissed):
Логических и стилистических недочетов не увидел.

NLazutkin committed:
rev. 1.9 Исправление ошибок в приложении в рамках подготовки к слиянию веток main и develop

- Обновлена ER-Диаграмма проекта
- Обновлено описание таблиц и примеры SQL запросов
- Рефакторинг методов классов FilmService и FilmStorageTest
- Исправление ошибок в методах InMemory для прохождения тестов

asyachem reviewed (approved):
все ок
